README.md

```diff
@@ -1,148 +1,213 @@
+depends on your host operation system. To increase the number start
+containers with the option `--ulimit nofile=65535:65535`:
 
-### Example Usage in a Multinode Setup
+## Multicast
 
-    HOSTS='crate1.example.com:4300,crate2.example.com:4300,crate3.example.com:4300'
-    HOST=crate1.example.com
-    docker run -d \
-        -p 4200:4200 \
-        -p 4300:4300 \
-        --name node1 \
-        --volume /mnt/data:/data \
-        --env CRATE_HEAP_SIZE=8g \
-        crate/crate:latest \
-        crate -Des.cluster.name=cratecluster \
+Crate uses multicast for node discovery by default. This means nodes
+started in the same multicast zone will discover each other
+automatically. However, Docker multicast support between containers on
+different hosts depends on the overlay network driver. If that does
+not support multicast, you have to [enable unicast in your custom
+`crate.yml`](https://crate.io/docs/reference/best_practice/multi_node_setup.html).
+
+Crate publishes the hostname it runs on for discovery within the
+cluster. If the address of the docker container differs from the
+actual host the docker image is running on -- this is the case if you
+do port mapping to the host via the `-p` option, you need to tell
+Crate to publish the address of the docker host instead:
+
+```console
+# docker run -d -p 4200:4200 -p 4300:4300 crate/crate \
+    crate -Des.network.publish_host=host1.example.com
+```
+
+If you change the transport port from the default `4300` to something
+else, you also need to pass the publish port to Crate by adding
+`-Des.transport.publish_port=4321` to your command.
+
+## Example Usage in a Multihost Setup
+
+To start a Crate cluster in containers distributed to three hosts
+without multicast enabled, run this command on the first node and
+adapt container and node names on the two other nodes:
+
+```console
+# HOSTS="crate1.example.com:4300,crate2.example.com:4300,crate3.example.com:4300"
+# HOST="crate1.example.com"
+# docker run -d -P \
+    --name crate1-container \
+    --volume /mnt/data:/data \
+    --env CRATE_HEAP_SIZE=8g \
+        crate/crate \
+	crate -Des.cluster.name=cratecluster \
               -Des.node.name=crate1 \
               -Des.transport.publish_port=4300 \
-              -Des.network.publish_host=$HOST \
+              -Des.network.publish_host="$HOST" \
               -Des.multicast.enabled=false \
-              -Des.discovery.zen.ping.unicast.hosts=$HOSTS \
+              -Des.discovery.zen.ping.unicast.hosts="$HOSTS" \
               -Des.discovery.zen.minimum_master_nodes=2
+```
 
 ## Crate Shell
 
-The Crate Shell (`crash`) is bundled with the Docker image. Since the `crash`
-executable is already in the `$PATH` environment variable, you can simply run:
+The Crate Shell `crash` is bundled with the Docker image. Since the
+`crash` executable is already in the `$PATH` environment variable,
+simply run:
+
+```console
+# docker run --rm -ti crate/crate crash --hosts [host1, host2, ...]
```

Comment:
No, don't think so. In this case no crate instance is started, so we don't need to map ports.